project/src/main/ui/level-select/career-level-library.gd

```diff
@@ -114,3 +114,72 @@ func _load_raw_json_data() -> void:
 			regions[i].length = regions[i + 1].distance - regions[i].distance
 		else:
 			regions[i].length = CareerData.MAX_DISTANCE_TRAVELLED
+
+
+## Returns a collection of chefs and customers required for potential upcoming cutscenes.
+##
+## Based on the upcoming cutscenes, we sometimes restrict the player's level choice to one or two levels where those
+## cutscenes make sense. This method returns the chefs/customers required for the upcoming cutscenes. As long as one
+## of these chefs/customers is selected by the player, there will be at least one appropriate cutscene which features
+## them.
+##
+## Returns:
+##	A dictionary with two entries:
+## 		'chef_ids' lists ids for creatures who acts as the chef for a cutscene.
+##
+## 		'customer_ids' lists ids for creatures who act as the main customer for a cutscene.
+func required_cutscene_characters() -> Dictionary:
+	var result := {
+		"chef_ids": [],
+		"customer_ids": [],
+	}
+	var region := region_for_distance(PlayerData.career.distance_travelled)
+	var potential_chat_key_pairs: Array = CareerCutsceneLibrary.potential_chat_key_pairs([region.cutscene_path])
+	for chat_key_pair in potential_chat_key_pairs:
+		for chat_key in chat_key_pair.chat_keys():
+			var chat_tree: ChatTree = ChatLibrary.chat_tree_for_key(chat_key)
+			if chat_tree.chef_id:
+				if not chat_tree.chef_id in result["chef_ids"]:
+					result["chef_ids"].append(chat_tree.chef_id)
+			elif chat_tree.customer_ids:
+				# If a cutscene specifies both a chef AND customer, we ignore the customer. The chef is the only
+				# creature who appears on the map.
+				if not chat_tree.customer_ids[0] in result["customer_ids"]:
+					result["customer_ids"].append(chat_tree.customer_ids[0])
+			else:
+				if not CareerLevel.ANONYMOUS_CUSTOMER in result["customer_ids"]:
+					result["customer_ids"].append(CareerLevel.ANONYMOUS_CUSTOMER)
+	return result
```

Comment:
We could use a chef_ids and customer_ids field which are assembled at the end. This would be a little less error prone (no fat-fingering magic strings)

Reply:
Added `chef_ids`, `customer_ids` variables which are assembled at the end.